src/main/scala/li/cil/oc/integration/opencomputers/DriverUpgradeTrading.scala

```diff
@@ -0,0 +1,37 @@
+package li.cil.oc.integration.opencomputers
+
+import li.cil.oc.api
+import li.cil.oc.Constants
+import li.cil.oc.api.driver.EnvironmentProvider
+import li.cil.oc.api.network.EnvironmentHost
+import li.cil.oc.api.driver.item.HostAware
+import li.cil.oc.common.{Tier, Slot}
+import li.cil.oc.server.component.UpgradeTrading
+import net.minecraft.item.ItemStack
+import li.cil.oc.common.entity.Drone
+import li.cil.oc.common.tileentity.Robot
+import li.cil.oc.server.component
+
+object DriverUpgradeTrading extends Item with HostAware {
+  override def worksWith(stack: ItemStack) = isOneOf(stack,
+    api.Items.get(Constants.ItemName.TradingUpgrade))
+
+  override def createEnvironment(stack: ItemStack, host: EnvironmentHost) =
+    if (host.world.isRemote) null
+    else host match {
+      case host: EnvironmentHost with Robot => new UpgradeTrading(host)
```

Comment:
I'd say the `with Robot` and `with Drone` here should go, since that will be handled via the blacklist (otherwise addon robots, should they be created, could not use this driver otherwise). I'll make the adjustment so you don't have to rebase again, if that's allright.